File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   val: number
 *   left: TreeNode | null
 *   right: TreeNode | null
 *   constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 *   }
 * }
 */
function maxDepth(root: TreeNode | null): number {
  if (!root) return 0
  let depth = 0
  const levels = {}
  levels[depth] = [root]
  while (levels[depth].length) {
    levels[depth + 1] = levels[depth + 1] || []
    levels[depth].forEach(node => {
      if (node.left) levels[depth + 1].push(node.left)
      if (node.right) levels[depth + 1].push(node.right)
    })
    depth++
  }
  return depth
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *   int val;
 *   TreeNode left;
 *   TreeNode right;
 *   TreeNode() {}
 *   TreeNode(int val) { this.val = val; }
 *   TreeNode(int val, TreeNode left, TreeNode right) {
 *     this.val = val;
 *     this.left = left;
 *     this.right = right;
 *   }
 * }
 */
class Solution {
  public int maxDepth(TreeNode root) {
    if (root == null) return 0;
    int depth = 0;
    List<List<TreeNode>> levels = new ArrayList();
    List<TreeNode> initial = new ArrayList<TreeNode>();
    initial.add(root);
    levels.add(initial);
    while (!((List<TreeNode>) levels.get(depth)).isEmpty()) {
      if (levels.size() < depth + 2) levels.add(new ArrayList<TreeNode>());
      List<TreeNode> curr = (List<TreeNode>) levels.get(depth);
      List<TreeNode> next = (List<TreeNode>) levels.get(depth + 1);
      curr.stream()
        .forEach(node -> {
          if (node.left != null) next.add(node.left);
          if (node.right != null) next.add(node.right);
        });
      depth++;
    }
    return depth;
  }
}